app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorEffectHandler.kt

```diff
@@ -38,19 +39,22 @@ class SyncIndicatorEffectHandler @AssistedInject constructor(
           .addTransformer(FetchDataForSyncIndicatorState::class.java, fetchDataForSyncIndicatorState())
           .addTransformer(StartSyncedStateTimer::class.java, startTimer())
           .addTransformer(InitiateDataSync::class.java, startDataSync())
-          .addTransformer(FetchPendingSyncRecordCount::class.java, fetchPendingSyncCount())
+          .addTransformer(FetchPendingSyncRecordsState::class.java, fetchPendingSync())
           .addConsumer(ShowDataSyncErrorDialog::class.java, { uiActions.showErrorDialog(it.errorType) }, schedulersProvider.ui())
           .build()
 
-  private fun fetchPendingSyncCount(): ObservableTransformer<FetchPendingSyncRecordCount, SyncIndicatorEvent> {
+  private fun fetchPendingSync(): ObservableTransformer<FetchPendingSyncRecordsState, SyncIndicatorEvent> {
     return ObservableTransformer { effect ->
       effect
-          .map {
-            frequentlySyncingRepositories
-                .map { it.pendingSyncRecordCountImmediate() }
-                .sum()
+          .flatMap {
+            val recordCounts = frequentlySyncingRepositories
+                .map { it.pendingSyncRecordCount() }
+
+            recordCounts
+                .combineLatest { counts -> counts.any { it > 0 } }
+                .filter { isSyncPending -> isSyncPending }
+                .map(::PendingSyncRecordsStateFetched)
```

Comment:
I was finding this a little hard to follow, so I tried rewriting it a little bit:

```kotlin
recordCounts
  .combineLatest { counts -> counts.sum() }
  .map { totalPendingSyncRecords -> totalPendingSyncRecords > 0 }
  .map(::PendingSyncRecordsStateFetched)
```

I felt we no longer needed a `.filter {}` here since the `Update` function is already checking the flag in the event that we emit. So the effect handler is only responsible for emitting the state of the pending sync records and the `Update` function is responsible for deciding what to do in the UI.

What do you think?